src/rust/pal/linux/mod.rs

```diff
@@ -61,3 +62,15 @@ pub unsafe fn set_nonblock(fd: RawFd) -> i32 {
     flags |= libc::O_NONBLOCK;
     libc::fcntl(fd, libc::F_SETFL, flags, 1)
 }
+
+/// Converts a [std::net::SocketAddrV4] into a [libc::sockaddr_in].
+pub fn socketaddrv4_to_sockaddr_in(addr: &SocketAddrV4) -> libc::sockaddr_in {
+    libc::sockaddr_in {
+        sin_family: libc::AF_INET as u16,
```

Comment:
I have no idea why libc defines `AF_INET` as a `libc::c_int` instead of as a `libc::sa_family_t`.  If they had, you wouldn't need this "`as`" clause.  But since you do need it, shouldn't it be "`as libc::sa_family_t`"?

Reply:
Thanks for noting this. `u16` was also working, but I agree that moving to the correct type cast is better.